code/game/objects/items/teleportation.dm

```diff
@@ -309,5 +309,180 @@
 			itemUser.visible_message(span_suicide("[user] looks even further depressed as they realize they do not have a head...and suddenly dies of shame!"))
 		return (BRUTELOSS)
 
+/obj/item/teleporter
+	name = "experimental teleporter"
+	desc = "A reverse-engineered version of the Nanotrasen portable handheld teleporter. Lacks the advanced safety features of its counterpart. A three-headed serpent can be seen on the back."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "syndi-tele"
+	throwforce = 5
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 4
+	throw_range = 10
+	flags_1 = CONDUCT_1
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	//Uses of the device left
+	var/charges = 4
+	//The maximum number of stored uses
+	var/max_charges = 4
+	///Minimum distance to teleport user forward
+	var/minimum_teleport_distance = 4
+	///Maximum distance to teleport user forward
+	var/maximum_teleport_distance = 8
+	//How far the emergency teleport checks for a safe position
+	var/parallel_teleport_distance = 3
+
+/obj/item/teleporter/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/teleporter/Destroy(force)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] has [charges] out of [max_charges] charges left.")
+
+/obj/item/teleporter/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return
+	attempt_teleport(user, FALSE)
+	return TRUE
+
+/obj/item/teleporter/process()
+	if(prob(10) && charges < max_charges)
+		charges++
+		if(ishuman(loc))
+			var/mob/living/carbon/human/holder = loc
+			to_chat(holder, span_notice("[src] beeps."))
+		playsound(src, 'sound/machines/twobeep.ogg', 10, TRUE, extrarange = SILENCED_SOUND_EXTRARANGE, falloff_distance = 0)
+
+/obj/item/teleporter/emp_act(severity)
+	var/teleported_something = FALSE
+	if(prob(50/severity))
+		if(ishuman(loc))
+			var/mob/living/carbon/human/user = loc
+			to_chat(user, span_danger("[src] buzzes!"))
+			attempt_teleport(user, TRUE)
+		else
+			var/turf/teleport_turf = get_turf(src)
+			for(var/mob/living/mob_on_same_tile in teleport_turf)
+				if(!teleported_something)
+					teleport_turf.visible_message(span_danger("[src] activates sporadically, teleporting everyone around it!"))
```

Comment:
Is this necessary? The visual feedback of everyone being teleported is already there, and I'm not sure this is something you'd be able to see so specifically.